packages/mui-base/src/SliderUnstyled/SliderUnstyled.tsx

```diff
@@ -274,53 +269,48 @@ const SliderUnstyled = React.forwardRef(function SliderUnstyled<
-                    : valueLabelFormat,
-                index,
-                open: open === index || active === index || valueLabelDisplay === 'on',
-                disabled,
-              })}
-              {...valueLabelProps}
-              className={clsx(classes.valueLabel, valueLabelProps.className)}
-            >
-              <Thumb
-                data-index={index}
-                data-focusvisible={focusedThumbIndex === index}
-                {...thumbProps}
-                className={clsx(classes.thumb, thumbProps.className, {
-                  [classes.active]: active === index,
-                  [classes.focusVisible]: focusedThumbIndex === index,
+          <Thumb
+            key={index}
+            data-index={index}
+            data-focusvisible={focusedThumbIndex === index}
+            {...thumbProps}
+            className={clsx(classes.thumb, thumbProps.className, {
+              [classes.active]: active === index,
+              [classes.focusVisible]: focusedThumbIndex === index,
+            })}
+            style={{
+              ...style,
+              pointerEvents: disableSwap && active !== index ? 'none' : undefined,
+              ...thumbProps.style,
+            }}
+          >
+            <Input
+              data-index={index}
+              aria-label={getAriaLabel ? getAriaLabel(index) : ariaLabel}
+              aria-valuenow={scale(value)}
+              aria-labelledby={ariaLabelledby}
+              aria-valuetext={
+                getAriaValueText ? getAriaValueText(scale(value), index) : ariaValuetext
+              }
+              value={values[index]}
+              {...inputProps}
+            />
+            {ValueLabel ? (
+              <ValueLabel
+                {...(!isHostComponent(ValueLabel) && {
+                  valueLabelFormat,
+                  index,
+                  disabled,
                 })}
-                style={{
-                  ...style,
-                  pointerEvents: disableSwap && active !== index ? 'none' : undefined,
-                  ...thumbProps.style,
-                }}
+                {...valueLabelProps}
               >
-                <Input
-                  data-index={index}
-                  aria-label={getAriaLabel ? getAriaLabel(index) : ariaLabel}
-                  aria-valuenow={scale(value)}
-                  aria-labelledby={ariaLabelledby}
-                  aria-valuetext={
-                    getAriaValueText ? getAriaValueText(scale(value), index) : ariaValuetext
-                  }
-                  value={values[index]}
-                  {...inputProps}
-                />
-              </Thumb>
-            </ValueLabelComponent>
-          </React.Fragment>
+                {typeof valueLabelFormat === 'function'
+                  ? valueLabelFormat(scale(value), index)
+                  : valueLabelFormat}
+              </ValueLabel>
+            ) : null}
```

Comment:
```suggestion
            ) : null}
            {thumbProps.children}
```

Shouldn't we also render the thumb's props.

Reply:
I don't understand. The thumb is now outside the value label.

Reply:
Based of the code in the component, the value label, together with the hidden input is rendered inside the thumb. I was wondering if there is a use-case where we would need to add the thumbProps.children too, for example if the tab needs to render some svg inside the dot.

Reply:
I would probably render the children first tough, not last :)

Reply:
> if the tab needs to render some svg inside the dot.

I think you mean the _slider_ needs to render some svg inside the dot, right? 

Well, I can't think of any such use case that would be needed.  If you have anything in mind, can you show a concrete example or an example of that kind of slider? We could add it once the community needs it. 

If they want to render, say an svg, it can probably be enough to provide a custom component through the thumb slot (slots.thumb).

Previously, when the input was inside the thumb, we didn't get any feature requests to support anything inside the thumb through slot.

Reply:
Alright, fair enough, let's skip it for now.